### PR TITLE
Bannerplugin typings

### PR DESCRIPTION
**Pull request by Legends**
Closing [issue](https://github.com/webpack/webpack/pull/7524) local branch is not available anymore.

**What kind of change does this PR introduce?**
no

**Did you add tests for your changes?**
no

**Does this PR introduce a breaking change?**
no

**What needs to be documented once your changes are merged?**
no

### CODE REVIEW

File: lib/BannerPlugin.js [hunk 1]
```diff
@@ -12,6 +12,23 @@ const Template = require("./Template");
 const validateOptions = require("schema-utils");
 const schema = require("../schemas/plugins/BannerPlugin.json");
 
+/** @typedef {import("./Compiler.js")} Compiler */
+
+/**
+ * BannerPlugin Options class.
+ * @typedef {Object=} BannerOptions
```
Comment (sokra) [inline lib/BannerPlugin.js]: `{Object=}` -> `{Object}`

Comment (Legends) [reply lib/BannerPlugin.js]: When I change it to `Object` TS gives me the following:

image

Comment (Legends) [reply lib/BannerPlugin.js]: What's this line for --> `this.options = options || {};` ?
There is a `validateOptions` call before, so I guess`undefined|null` is not valid for `options`...

File: lib/BannerPlugin.js [hunk 2]
```diff
@@ -80,7 +80,6 @@ class BannerPlugin {
 	/**
 	 * Apply the plugin
 	 * @param {Compiler} compiler Webpack Compiler
-	 * @returns {void}
```
Comment (Legends) [inline lib/BannerPlugin.js]: https://github.com/webpack/webpack/issues/7122 - section Return!?

**Comment by evilebottnawi** [comment]
Fixed in webpack@5 https://github.com/webpack/webpack/blob/master/declarations/plugins/BannerPlugin.d.ts, anyway if you have improvements - PR welcome

**Pull request closed without merging**